pkg/auth/metadata.go

```diff
@@ -170,10 +171,15 @@ func (m *KVMetadataManager) writeMetadata(ctx context.Context, items map[string]
 	return nil
 }
 
-func (m *KVMetadataManager) UpdateSetupTimestamp(ctx context.Context, ts time.Time) error {
-	return m.writeMetadata(ctx, map[string]string{
-		SetupTimestampKeyName: ts.UTC().Format(time.RFC3339),
-	})
+func (m *KVMetadataManager) UpdateSetupTimestamp(ctx context.Context, setupTime time.Time, authType string) error {
+	setupTimeStr := setupTime.UTC().Format(time.RFC3339)
+	items := map[string]string{
+		SetupTimestampKeyName: setupTimeStr,
+	}
+	if authType != "" {
+		items[SetupAuthTypeKeyPrefix+authType] = setupTimeStr
+	}
```

Comment:
I don't think that `authType` is empty, it always has value. 
```suggestion
	items[SetupAuthTypeKeyPrefix+authType] = setupTimeStr
```